### PR TITLE
required mail server in discourse

### PR DESCRIPTION
**Pull request by rawdaGastan**
### Description

- Memory, disk and cpu validations are already done.
- Added new validations for the mail server fields in discourse to be required. As deployment can't proceed without it.

### Changes

- added a new validator for host name, port, username and password in mail server of discourse

### Related Issues

#795 

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings

### CODE REVIEW

File: src/utils/validateDomainName.ts [hunk 1]
```diff
@@ -7,3 +7,9 @@ export default function validateDomainName(domain: string): string | void {
     if (domain === "") return null;
     if (!DOMAIN_NAME_REGEX.test(domain)) return "Domain name is not valid";
 }
+
+export function validateRequiredHostName(domain: string): string | void {
+    // for optional fields to work
+    if (domain === "") return "Host name is required";
+    if (!DOMAIN_NAME_REGEX.test(domain)) return "Host name is not valid";
+}
```
Comment (AhmedHanafy725) [inline src/utils/validateDomainName.ts:15]: Please use `validateDomainName` here instead of re-implement it

File: src/utils/validateName.ts [hunk 2]
```diff
@@ -26,6 +26,12 @@ export function validateEmail(email: string): string | void {
   if (!EMAIL_REGEX.test(email)) return "Invalid email format.";
 }
 
+export function validateRequiredEmail(email: string): string | void {
+  if (email == "") return "Email is required";
+  if (!EMAIL_REGEX.test(email)) return "Invalid email format.";
+}
```
Comment (AhmedHanafy725) [inline src/utils/validateName.ts:32]: here too

File: src/utils/validateName.ts [hunk 3]
```diff
@@ -83,6 +89,15 @@ export function validatePortNumber(value: string): string | void {
...
 }
 
+export function validateRequiredPortNumber(value: string): string | void {
+  if (value === "") return "Port is required.";
+  let port = +value;
+  if (isNaN(port)) return "Port must be a valid number.";
+  if (port.toFixed(0) !== value) return "Port must be a valid integer.";
+  if (port < 1) return "Minimum allowed port is 1.";
+  if (port > 65535) return "Maximum allowed port is 65535.";
+}
```
Comment (AhmedHanafy725) [inline src/utils/validateName.ts:99]: Please use `validatePortNumber` here instead of re-implement it

File: src/utils/validateName.ts [hunk 4]
```diff
@@ -101,6 +116,12 @@ export function validatePassword(value: string): string | void {
   if (value.length > 15) return "Password must be at least 15 characters";
 }
 
+export function validateRequiredPassword(value: string): string | void {
+  if (value == "") return "Password is required";
+  if (value.length < 6) return "Password must be at least 6 characters";
+  if (value.length > 15) return "Password must be at least 15 characters";
+}
+
```
Comment (AhmedHanafy725) [inline src/utils/validateName.ts:124]: Please use `validatePassword` here instead of re-implement it

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: src/utils/validateDomainName.ts (hunk 1 above)
Comment (rawdaGastan) [reply src/utils/validateDomainName.ts:15]: ValidateDomainName has a different error --> Domain name instead of Host name

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**